core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/IngestUtils.java

```diff
@@ -10,13 +10,82 @@
  ******************************************************************************/
 package mil.nga.giat.geowave.core.ingest;
 
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.net.URLStreamHandlerFactory;
 import java.util.List;
 
+import mil.nga.giat.geowave.core.ingest.hdfs.HdfsUrlStreamHandlerFactory;
+import mil.nga.giat.geowave.core.ingest.s3.S3URLStreamHandlerFactory;
 import mil.nga.giat.geowave.core.store.index.CommonIndexValue;
 import mil.nga.giat.geowave.core.store.operations.remote.options.IndexPluginOptions;
 
 public class IngestUtils
 {
+	public static enum URLTYPE {
+		S3,
+		HDFS
+	}
+
+	public static void setURLStreamHandlerFactory(
+			URLTYPE urlType )
+			throws NoSuchFieldException,
+			SecurityException,
+			IllegalArgumentException,
+			IllegalAccessException {
+
```

Comment:
this would be a good place for a static boolean check (per URLType) to make sure we aren't setting it more than we have to